systemtest/src/main/java/io/strimzi/systemtest/apiclients/MsgCliApiClient.java

```diff
@@ -0,0 +1,233 @@
+        VertxOptions options = new VertxOptions()
+                .setWorkerPoolSize(1)
+                .setInternalBlockingPoolSize(1)
+                .setEventLoopPoolSize(1);
+        this.vertx = Vertx.vertx(options);
+        this.connect();
+    }
+
+    private String apiClientName() {
+        return "Kafka Clients";
+    }
+
+    protected void connect() {
+        this.webClient = WebClient.create(vertx, new WebClientOptions()
+                .setSsl(false)
+                .setTrustAll(true)
+                .setVerifyHost(false));
+    }
+
+    protected <T> T doRequestNTimes(int retry, Callable<T> fn, Optional<Runnable> reconnect) throws Exception {
+        return TestUtils.doRequestNTimes(retry, fn, reconnect);
+    }
+
+    private <T> void responseHandler(AsyncResult<HttpResponse<T>> ar, Future<T> promise, int expectedCode,
+                                     String warnMessage) {
+        try {
+            if (ar.succeeded()) {
+                HttpResponse<T> response = ar.result();
+                T body = response.body();
+                if (response.statusCode() != expectedCode) {
+                    LOGGER.error("expected-code: {}, response-code: {}, body: {}", expectedCode, response.statusCode(), response.body());
+                    promise.fail(new RuntimeException("Status " + response.statusCode() + " body: " + (body != null ? body.toString() : null)));
+                } else if (response.statusCode() < HTTP_OK || response.statusCode() >= HttpURLConnection.HTTP_MULT_CHOICE) {
+                    promise.fail(new RuntimeException(body.toString()));
+                } else {
+                    promise.complete(ar.result().body());
+                }
+            } else {
+                LOGGER.warn(warnMessage);
+                promise.fail(ar.cause());
+            }
+        } catch (io.vertx.core.json.DecodeException decEx) {
+            if (ar.result().bodyAsString().contains("application is not available")) {
+                LOGGER.warn("'{}' is not available.", apiClientName(), ar.cause());
+                throw new IllegalStateException(String.format("'%s' is not available.", apiClientName()));
+            } else {
+                LOGGER.warn("Unexpected object received", ar.cause());
+                throw new IllegalStateException("JsonObject expected, but following object was received: " + ar.result().bodyAsString());
+            }
+        }
+    }
+
+    /**
+     * Start new messaging kafka client(s)
+     *
+     * @param clientArguments list of arguments for kafka client (together with kafka client name!)
+     * @param count           count of clients that will be started
+     * @return
+     * @throws InterruptedException
+     * @throws ExecutionException
+     * @throws TimeoutException
+     */
+    public JsonObject startClients(List<String> clientArguments, int count) throws Exception {
+        Future<JsonObject> responsePromise = Future.future();
```

Comment:
You need to move this line inside the lambda you're passing to `doRequestNTimes()`, because otherwise the recursive calls to `doRequestNTimes()` will end up trying to complete the future _again_ (which is not allowed. `Futures` are one-shot).